app/javascript/components/patient/laboratory/Laboratory.js

```diff
@@ -44,29 +48,88 @@ class Laboratory extends React.Component {
     });
   };
 
+  toggleDeleteModal = () => {
+    let current = this.state.showDeleteModal;
+    this.setState({
+      showDeleteModal: !current,
+      delete_reason: null,
+      delete_reason_text: null,
+    });
+  };
+
+  handleDeleteSubmit = () => {
+    let deleteReason = this.state.delete_reason;
+    if (deleteReason === 'Other' && this.state.delete_reason_text) {
+      deleteReason += ', ' + this.state.delete_reason_text;
+    }
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .delete(window.BASE_PATH + '/laboratories/' + this.props.lab.id, {
+          data: {
+            patient_id: this.props.patient.id,
+            delete_reason: deleteReason,
+          },
+        })
+        .then(() => {
+          this.setState({ loading: false, showDeleteModal: false });
+          location.reload(true);
+        })
+        .catch(error => {
+          this.setState({ loading: false, showDeleteModal: false });
+          reportError(error);
+        });
+    });
+  };
+
   render() {
     return (
       <React.Fragment>
         {!this.props.lab.id && (
-          <Button onClick={() => this.setState({ showModal: true, loading: false })}>
+          <Button onClick={this.toggleLabModal}>
             <i className="fas fa-plus fa-fw"></i>
             <span className="ml-2">Add New Lab Result</span>
           </Button>
         )}
         {this.props.lab.id && (
-          <Button variant="link" onClick={() => this.setState({ showModal: true, loading: false })} className="btn btn-link py-0" size="sm">
-            <i className="fas fa-edit"></i> Edit
-          </Button>
+          <Dropdown>
+            <Dropdown.Toggle
+              id={`vaccine-action-button-${this.props.lab.id}`}
+              size="sm"
+              variant="primary"
+              aria-label={`vaccine-action-button-${this.props.lab.id}`}>
```

Comment:
Same with this aria-label.  It should probably say "Lab Results Action Dropdown"